### ISSUE TITLE
Loss value records for unseen test set

### ISSUE THREAD

**Issue by dalbom** [KIST-Iceberg/Iceberg#5]
Kaggle Leader Board 결과값이 아닌, 트레인 셋에서 임의로 뽑은 테스트 셋에 대한 loss 값 랭킹.

소수점 이하 다섯째 자리에서 반올림.

Keras의 Dropout Rate와 TensorFlow의 Keep Prob이 서로 반대되는 의미라 TensorFlow 기준으로 수정.

디폴트 세팅에서 변화를 준 부분 혹은 특이사항에 볼드 페이스 처리.

디폴트 세팅:

> Learning rate: **0.001**
> Keep Prob: **0.8**
> Linear unit: **ReLU**
> Triple the train set using flips: **Yes**
> Preprocessing using high-pass filtering and gamma correction: **Yes**

### Current best LB Loss: 0.1418

- Ensembling 7 submissions
- Rounded prediction results for values > 0.98 and < 0.02

### Test Loss Ranking
- 0.1635 
  - Train loss: 0.1241
  - learning rate: 0.0001
  - using SR model
  - epoch 5
  - batch 30
  - **Keep Prob: 0.9**
  - **Batch normalization**
  - **Removed NaN data**
  - Test summit

- 0.1763 (LB Loss: 0.1916)
   - Train loss: 0.0994
   - **Combined channel - |HH|*|HV|**
   - **Batch normalization**
   - **Removed NaN data**
   - Learning rate: **exp_range (base_lr=0.001, max_lr=0.006, step_size=230., gamma=0.99994)**

- 0.1906 (LB Loss: 0.2002)
   - Train loss: 0.1341
   - **Batch normalization**
   - **Removed NaN data**
   - Learning rate: **exp_range (base_lr=0.001, max_lr=0.006, step_size=230., gamma=0.99994)**

- 0.2063
   - Train loss: 0.1708
   - **Removed NaN data**
   - Learning rate: **exp_range (base_lr=0.001, max_lr=0.006, step_size=230., gamma=0.99994)**

- 0.2613 (LB Loss: 0.2244)
   - Train loss: 0.1711
   - Default Setting

- 0.2660
   - Train loss: 0.1483
   - Learning rate: **exp_range (base_lr=0.001, max_lr=0.006, step_size=230., gamma=0.99994)**

- 0.2700
   - Train loss: 0.1315
   - Keep Prob: **0.78**

- 0.2823
   - Train loss: 0.2535
   - Keep Prob: **0.9**

- 0.2872
   - Train loss: 0.2149
   - Learning rate: **exp_range (base_lr=0.001, max_lr=0.006, step_size=230., gamma=0.99994)**
   - Linear unit: **ELU**

- 0.2913
   - Train loss: 0.2576
   - Keep Prob: **0.7**

- 0.3081
   - Train loss: 0.1164
   - Learning rate: **0.002**
   - Linear unit: **ELU**

- 0.3237
   - Train loss: 0.1858
   - Triple the train set using flips: **No**

- 0.3277
   - Train loss: **0.0683 - Lowest train loss**
   - Linear unit: **ELU**
   
- 0.4584
   - Train loss: 0.4039
   - Learning rate: **0.0002**
   - Linear unit: **ELU**

**Comment by mike2ox** [comment]
I think that how to decrease loss value

- Learning rate : fixed ->Cyclical Learning Rate 
(https://arxiv.org/pdf/1506.01186.pdf)
(http://teleported.in/posts/cyclic-learning-rate/)
(https://github.com/bckenstler/CLR)
- activation function : only ReLU -> mix activation(ex, ReLU + Leaky ReLU) (?)
- apply object detection(except sea surface)

**Comment by mike2ox** [comment]
기본 제공되는 polarized data(HH,HV)에서 새로운 data를 도출

> 1. https://earth.esa.int/c/document_library/get_file?folderId=409229&name=DLFE-5566.pdf (2013년 유럽우주국)
>~2. https://www.the-cryosphere.net/11/33/2017/ (2017년 The Cryosphere)~ 논문에서 공식 적용이 어려움.
